### PR TITLE
Pbi 258 1

### PR DESCRIPTION
**Pull request by pilievOoyala**
*I created the gulp file to build the ad managers. 
* You can use "make test" to run the unit tests. 
*I took out vast and it's tests for the moment. I'll need to eval how much work it will be to get it to work without it's dependencies on OO internals. 
*Added gitignore file.
*took out freewheel integration tests. For now they are dependent on external files. Will create a submodule to resolve this in the next checkin.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+node_modules
+build
+coverage
```
Comment (angieisrani) [inline .gitignore]: I don't think we use this yet - might not make sense until or unless we add coverage testing

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+var getFileNameFromPath = function(path)
+{
+  var start = path.lastIndexOf('/') + 1;
+  return path.substring(start);
+}
+
+// Run tests
+gulp.task('test', shell.task(['npm test']));
```
Comment (angieisrani) [inline gulpfile.js]: have this point to make test

File: js/ad_manager_vast.js [hunk 3]
```diff
@@ -310,13 +324,12 @@ OO.Ads.manager((function(_, $) {
...
-        this.amc.showSkipVideoAdButton();
-        var clickUrl = adWrapper.ad.data.linear.ClickThrough.length > 0 ?
-                       adWrapper.ad.data.linear.ClickThrough[0] : null;
+        this.amc.showSkipVideoAdButton(true);
+        var hasClickUrl = adWrapper.ad.data.linear.ClickThrough.length > 0;
         this.amc.notifyLinearAdStarted(this.name, {
             name: adWrapper.ad.data.title,
             duration: adWrapper.ad.durationInMilliseconds/1000,
-            clickUrl: clickUrl,
+            hasClickUrl: true,
```
Comment (angieisrani) [inline js/ad_manager_vast.js]: consider separating this into another commit

File: scripts/test_lib.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+global.SRC_ROOT = '../../js/';
```
Comment (angieisrani) [inline scripts/test_lib.js]: consider moving to test directory

File: scripts/test_lib.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+global.window.setTimeout = setTimeout;
+global.navigator = window.navigator;
+
+global.expect = require('expect.js');
+
+global.OO._ = window._ = global._ = require("underscore");
+global.OO.$ = window.$ = global.$ = require("jquery");
+
+require('hazmat');  // hazmat detects this as browser because of window stuff
+global.OO.HM  = global.window.Hazmat.create();
```
Comment (angieisrani) [inline scripts/test_lib.js]: I wonder if you may be able to remove these settings based on the loading of libraries from the utils repo

File: test/unit-tests/vast_test.js [hunk 6]
```diff
@@ -1,518 +0,0 @@
-/*
```
Comment (angieisrani) [inline test/unit-tests/vast_test.js]: You will be able to bring this file back now that vast is in the repo

**Comment by pilievOoyala** [comment]
Greg approves. And the last couple of commits are to update the ad managers to the latest code in mjolnir. I retested and everything works as expected.

**Pull request merged**